src/schema/v2/notification_preferences.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  GraphQLEnumType,
+  GraphQLNonNull,
+  GraphQLFieldConfig,
+  GraphQLList,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLString,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+
+const subGroupfields = {
+  id: {
+    type: GraphQLString,
```

Comment:
So I'm not sure what the difference between input and output in terms of requirement but — should any of these other fields also be non-nullable? Presumably at least ID

Reply:
Yeah all these things will be non-null, updated with [132d848](https://github.com/artsy/metaphysics/pull/3708/commits/132d848b6539e4f4e14a9d9b25fa200cd00dee5f). Note technically all Gravity cares about in the POST endpoint is the status/name duo and the rest are ignored but providing the rest doesn't hurt anything.